src/libtorch.cc

```diff
@@ -1146,21 +1238,36 @@ ModelInstanceState::ProcessRequests(
   std::vector<torch::jit::IValue> input_tensors;
   bool cuda_copy = false;
   std::unique_ptr<BackendInputCollector> collector;
+  std::mutex timestamp_mu;
+
+  uint64_t compute_input_start = 0;
+  std::tuple<uint64_t*, std::mutex*> compute_input_cb_data(
+      &compute_input_start, &timestamp_mu);
+
   if (Kind() == TRITONSERVER_INSTANCEGROUPKIND_GPU) {
 #ifdef TRITON_ENABLE_GPU
     RESPOND_ALL_AND_SET_TRUE_IF_ERROR(
         responses, request_count, all_response_failed,
         ConvertCUDAStatusToTritonError(
             cudaEventRecord(compute_input_start_event_, stream_),
             TRITONSERVER_ERROR_INTERNAL, "Failed to record the event."));
+#endif
+  } else if (Kind() == TRITONSERVER_INSTANCEGROUPKIND_MODEL) {
+#ifdef TRITON_ENABLE_GPU
+    for (const auto& stream : stream_vec_) {
+      cudaLaunchHostFunc(
+          stream, CaptureFirstTimestampCallback,
+          reinterpret_cast<void*>(&compute_input_cb_data));
+    }
```

Comment:
As discussed, I don't think for inputs we need to make any changes. We only need the multi-stream logic for calculating the `compute infer` time.

Reply:
Removed this part and use cuda events for the comput input duration.